api/main.py

```diff
@@ -97,8 +97,10 @@ async def post_node(node: Node, token: str = Depends(get_user)):
             obj = await db.create(node)
             op = 'created'
         else:
-            obj = await db.update(node)
-            op = 'updated'
+            # If node id is passed in request JSON
```

Comment:
The `Database.create()` method already raises an error if the id is not None:
```python
        if obj.id is not None:
            raise ValueError(f"Object cannot be created with id: {obj.id}")
```
So I think we can just remove the `if`/`else` logic from here, and let the `Database` class raise a `ValueError` when there is a problem.  That will then result in `HTTP 403` when the exception is caught below.